engine/src/test/java/io/camunda/zeebe/engine/processing/bpmn/gateway/ExclusiveGatewayTest.java

```diff
@@ -365,4 +367,141 @@ public void shouldResolveIncidentsWhenTerminating() {
         .extracting(Record::getIntent)
         .containsExactly(IncidentIntent.CREATED, IncidentIntent.RESOLVED);
   }
+
+  @Test
+  public void shouldCreateDeploymentExclusiveGatewayWithDefaultFlow() {
+    // given
+    final String processId = Strings.newRandomValidBpmnId();
+    // when
+    final BpmnModelInstance processDefinition1 =
+        Bpmn.createExecutableProcess(processId)
+            .startEvent()
+            .exclusiveGateway("xor")
+            .sequenceFlowId("s1")
+            .conditionExpression("= contains(str,\"a\")")
+            .endEvent("end1")
+            .moveToLastExclusiveGateway()
+            .defaultFlow()
+            .sequenceFlowId("s2")
+            .endEvent("end2")
+            .done();
+
+    final Record<DeploymentRecordValue> deployment1 =
+        ENGINE.deployment().withXmlResource(processDefinition1).deploy();
+
+    // then
+    assertThat(deployment1.getKey())
+        .describedAs("Exclusive gateway's default flow should be allowed to have no condition")
+        .isNotNegative();
+
+    // when
+    final BpmnModelInstance processDefinition2 =
+        Bpmn.createExecutableProcess(processId)
+            .startEvent()
+            .exclusiveGateway("xor")
+            .sequenceFlowId("s1")
+            .conditionExpression("= contains(str,\"a\")")
+            .endEvent("end1")
+            .moveToLastExclusiveGateway()
+            .defaultFlow()
+            .sequenceFlowId("s2")
+            .conditionExpression("= contains(str,\"b\")")
+            .endEvent("end2")
+            .done();
+
+    final Record<DeploymentRecordValue> deployment2 =
+        ENGINE.deployment().withXmlResource(processDefinition2).deploy();
+
+    // then
+    assertThat(deployment2.getKey())
+        .describedAs(
+            "Exclusive gateway's default flow should be allowed to have a condition, but not be required")
+        .isNotNegative();
+  }
+
+  @Test
+  public void shouldNotEvaluateConditionOfDefaultFlow() {
+    // given
+    final String processId = Strings.newRandomValidBpmnId();
+    final BpmnModelInstance processDefinition =
+        Bpmn.createExecutableProcess(processId)
+            .startEvent()
+            .exclusiveGateway("xor")
+            .sequenceFlowId("s1")
+            .conditionExpression("= contains(str,\"a\")")
+            .endEvent("end1")
+            .moveToLastGateway()
+            .sequenceFlowId("s2")
+            .conditionExpression("= contains(str,\"b\")")
+            .endEvent("end2")
+            .moveToLastExclusiveGateway()
+            .defaultFlow()
+            .sequenceFlowId("s3")
+            .conditionExpression("= true")
```

Comment:
❌  The assertion below `Expect that the default flow's condition is not evaluated` verifies that no incident is created. That is only relevant if the expression would create an incident when evaluated. Currently, it just evaluates to `true` and never creates an incident.

We should change the condition expression to `= nonexisting_variable` to make sure an incident would be created when this condition would be evaluated.

Reply:
> We should change the condition expression to = nonexisting_variable to make sure an incident would be created when this condition would be evaluated.

❓  Hi @korthout. I thought you may have missed something? 

According to the description in the specification
> The default Sequence Flow should not have a conditionExpression. Any such Expression SHALL be ignored.

When `default sequence Flow` has conditionExpression we should ignore it, so I thought we would not evaluate the expression and we never create the event. Please correct me if I understand wrong. ❤️

Reply:
>When default sequence Flow has conditionExpression we should ignore it, so I thought we would not evaluate the expression and we never create the event. Please correct me if I understand wrong

That's correct, but we should verify that we don't evaluate it in a test. We do that by using an expression that we expect would create an incident in case it is evaluated. We can then verify that no incident is created. This tells us that the expression was not evaluated. Does that make sense?